src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs

```diff
@@ -233,15 +233,40 @@ public bool ShouldHandleScheme(string authenticationScheme, bool handleAutomatic
             }
         }
 
+        /// <summary>
+        /// Handle the authentication for once.
+        ///
+        /// If the authentication has been done before returns the last authentication result.
+        /// </summary>
         protected Task<AuthenticateResult> HandleAuthenticateOnceAsync()
         {
             if (_authenticateTask == null)
             {
                 _authenticateTask = HandleAuthenticateAsync();
             }
+
             return _authenticateTask;
         }
 
+        /// <summary>
+        /// Handle the authentication for once.
+        ///
+        /// If the authentication has been done before returns the last authentication result.
```

Comment:
Maybe:

Used to ensure HandleAuthenticateAsync is only invoked once safely, subsequent calls will return the same authentication result.  Any exceptions will be converted into a failed AuthenticationResult containing the exception.